mrmustard/lab_dev/states/base.py

```diff
@@ -618,6 +642,35 @@ def __rshift__(self, other: CircuitComponent) -> CircuitComponent:
     def __repr__(self) -> str:
         return ""
 
+    def __getitem__(self, modes: Union[int, Sequence[int]]) -> State:
```

Comment:
This should have been the initial design. Let's hope it will be a well-received change!